packages/react-components/src/components/chart/baseChart.tsx

```diff
@@ -36,7 +33,10 @@ const keyMap: KeyMap = {
  * 2. get datastreams using useVisualizedDataStreams
  * 3. use datastreams / chart options to compute datastructures
  *    needed to implement various features and adapt them to the echarts api
- * 4. set all of the options in echarts using useEChartOptions
+ * 4. set all  the options in echarts using useEChartOptions
+ * 5. do not make use of setOptions in the individual feature, useEChartOptions should be the only place.
+ *    Exception: when deleting an item or in general when removing some elements you may need to use setOptions
```

Comment:
We don't need to use this exception if we correctly set the settings for setOption. I think we should try not to use the echarts ref to do this outside of useEChartOptions

Reply:
right, thats why its an exception :)